plugin/pkg/shoot/validator/admission.go

```diff
@@ -354,6 +354,18 @@ func (c *validationContext) validateProjectMembership(a admission.Attributes) er
 	return nil
 }
 
+// validateSeedSelectionForHAShoot validates the seed assigned for the shoot based on the following criteria:
+// 1. A HA shoot with failure tolerance of 'zone' can only be scheduled on multi-zonal seeds
+// 2. A HA shoot with failure tolerance of 'node' or a non-HA shoot can be scheduled onto a multi-zonal as well as non-multi-zonal seed.
+func (c *validationContext) validateSeedSelectionForHAShoot() error {
```

Comment:
Don't we also need this type of validation in `plugin/pkg/shoot/binding/admission.go` or was is discussed otherwise?

Reply:
No we do not. See https://github.com/gardener/gardener/pull/6533/files#r961882424
I originally planned to have this done only in #6533 but then i saw that Shreyas had already included changes in this fine as part of this PR so i brought this in here.

Essentially shoot validator is called for both scheduled and re-scheduled (CPM) case. So it is going to work fine even for rescheduling work flows.

Reply:
Okay, thanks. Shouldn't we then only validate if `shootIsBeingScheduled || shootIsBeingRescheduled` as the comment suggests?

Reply:
Added